django_log_formatter_asim/__init__.py

```diff
@@ -1,177 +1,95 @@
-                message=self.record.getMessage(),
-                labels={
-                    **labels,
-                    **extra_labels,
-                },
-            )
-            .host(
-                architecture=platform.machine(),
-            )
-        )
-
-        return logger
-
-    def _get_event_category(self):
-        if self.record.name in ("django.request", "django.server"):
-            return CATEGORY_WEB
-        if self.record.name.startswith("django.db.backends"):
-            return CATEGORY_DATABASE
-
-        return CATEGORY_PROCESS
-
-    def _get_environment(self):
-        return os.getenv("DJANGO_SETTINGS_MODULE") or "Unknown"
+        return log_dict
+
+    def _get_event_severity(self, log_level):
+        map = {
+            "DEBUG": "Informational",
+            "INFO": "Informational",
+            "WARNING": "Low",
+            "ERROR": "Medium",
+            "CRITICAL": "High",
+        }
+        return map[log_level]
 
 
 class ASIMSystemFormatter(ASIMFormatterBase):
-    def get_event(self):
-        logger_event = self._get_event_base()
-
-        return logger_event
-
-
-class ASIMDBFormatter(ASIMFormatterBase):
-    # created for augmentation based on django.db.backends
-    def get_event(self):
-        logger_event = self._get_event_base()
-
-        return logger_event
+    def get_log_dict(self):
+        return self._get_log_dict_base()
 
 
 class ASIMRequestFormatter(ASIMFormatterBase):
-    def get_event(self):
-        zipkin_headers = getattr(
-            settings,
-            "DLFE_ZIPKIN_HEADERS",
-            ("X-B3-TraceId", "X-B3-SpanId"),
-        )
-
-        extra_labels = {}
-
-        for zipkin_header in zipkin_headers:
-            if getattr(
-                self.record.request.headers,
-                zipkin_header,
-                None,
-            ):
-                extra_labels[zipkin_header] = self.record.request.headers[
-                    zipkin_header
-                ]  # noqa E501
+    def get_log_dict(self):
+        log_dict = self._get_log_dict_base()
+
+        request = self.record.request
+
+        # Source fields...
+        log_dict["SrcIpAddr"] = request.environ.get("REMOTE_ADDR", None)
+        log_dict["IpAddr"] = log_dict["SrcIpAddr"]
+        log_dict["SrcPortNumber"] = request.environ.get("SERVER_PORT", None)
+
+        # Acting Application fields...
+        http_user_agent = getattr(request, "user_agent", None)
+        if not http_user_agent:
+            http_user_agent = getattr(request.headers, "user_agent", None)
+        if not http_user_agent:
+            http_user_agent = request.META.get("HTTP_USER_AGENT", None)
+        log_dict["HttpUserAgent"] = http_user_agent
+
+        user = getattr(request, "user", None)
+        user_id = None
+        username = None
+        if user:
+            user_id = getattr(user, "id", None)
+            if user.__class__.__name__ == "AnonymousUser":
```

Comment:
Can we not do the following?

```suggestion
            if user.is_anonymous:
```

Reply:
Done